File: codingTest/baekjoon/javascript/1158.js
const rl = require("readline").createInterface({ input: process.stdin, output: process.stdout });

const input = [];
rl.on("line", line => {
  input.push(line);
}).on("close", () => {
  const [N, K] = input
    .shift()
    .split(" ")
    .map(e => +e);
  const arr = new Array(N).fill(1).map((e, i) => e + i);
  const result = [];
  let i = 0;
  while (arr.length != 0) {
    i = (i + K - 1) % arr.length;
    result.push(arr.splice(i, 1)[0]);
  }
  console.log("<" + result.toString().replace(/,/g, ", ") + ">");
});

File: codingTest/baekjoon/python/1158.py
from collections import deque
N, K = map(int ,input().split())
arr = deque([i for i in range(1, N+1)])
result = []
idx = 0
while arr:
    if idx == K - 1: 
        result.append(arr.popleft())
        idx = 0
    else:
        idx += 1
        arr.append(arr.popleft())
print("<" + ", ".join(map(str, result)) + ">")
